v1.1/demo-follow-the-workload.md

```diff
@@ -0,0 +1,298 @@
+---
+title: Follow the Workload
+summary: CockroachDB can dynamically optimize read latency for the location from which most of the workload is originating.
+toc: false
+---
+
+"Follow the workload" refers to CockroachDB's ability to dynamically optimize read latency for the location from which most of the workload is originating. This page describes how this works and walks you through a simple demonstration using a local cluster.
+
+<div id="toc"></div>
+
+## Overview
+
+### Basic Terms
+
+To understand how "follow the workload" works, it's important to start with some basic terms:
+
+Term | Description
+-----|------------
+**Range** | CockroachDB stores all user data and almost all system data in a giant sorted map of key-value pairs. This keyspace is divided into "ranges", contiguous chunks of the keyspace, so that every key can always be found in a single range.
+**Range Replica** | CockroachDB replicates each range (3 times by default) and stores each replica on a different node.
+**Range Lease** | For each range, one of the replicas holds the "range lease". This replica, referred to as the "leaseholder", is the one that receives and coordinates all read and write requests for the range.
+
+### How It Works
+
+"Follow the workload" is based on the way **range leases** handle read requests. Read requests bypass the Raft consensus protocol, accessesing the range replica with which the range lease is co-located (the leaseholder) and sending the results to the client without needing to coordinate with any of the other range replicas. Bypassing Raft, and the network roundtrips involved, is possible because the leaseholder is guaranteed to be up-to-date already due to the way write request are handled by Raft.
+
+This increases the speed of reads, but it doesn't guarantee that the range lease will be anywhere close to the origin of requests. If requests are coming from the US West, for example, and the relevant range lease is on a node in the US East, the requests would likely enter a gateway node in the US West and then get routed to the node with the range lease in the US East.
+
+However, you can cause the cluster to actively move range leases for even better read performance by starting each node with the `--locality` flag. With this flag specified, the cluster knows about the location of each node, so when there's high latency between nodes, the cluster will move active range leases to a node closer to the origin of the majority of the workload. This is especially helpful for applications with workloads that move around throughout the day (e.g., most of the traffic is in the US East in the morning and in the US West in the evening).
```

Comment:
Nit: "--locality flag" can be hyperlinked to the locality flag documentation? Also, the locality flag documentation can be updated to reference this document.